src/engine/.rtr/rtrtests.py

```diff
@@ -0,0 +1,488 @@
+                Uses 'params.source' to get the path to the source folder.
+                Uses 'params.exclude' to ignore folders from scan.
+
+    Returns:
+        bool: True on success, False otherwise.
+    """
+    command = 'cppcheck'
+    # Creating folder for cppcheck build
+    os.makedirs(os.path.join(params.output, 'build', 'cppcheck-build'), exist_ok=True)
+    args = f'--error-exitcode=1 --force {params.source}'
+    args += f' --std=c++17 --enable=warning,style,performance,portability,unusedFunction --suppress=constParameterCallback '
+    args += f' --cppcheck-build-dir={params.output}{BUILDDIR}/cppcheck-build '
+    if params.exclude:
+        abs_ignore = [os.path.join(params.source, path)
+                      for path in params.exclude]
+        args += f'-i {" -i ".join(abs_ignore)}'
+    logging.debug(f'Executing {command} {args}')
+    result = subprocess.run(
+        f'{command} {args}', stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if result.returncode == 0 and not result.stderr:
+        logging.info('CPPCHECK: successful')
+    else:
+        logging.info('CPPCHECK: fail')
+        # TODO: we force the return code to 0 to allow the tool to continue
+        result.returncode = 0
+    log(params.output, 'cppcheck', result.stdout, result.stderr)
+    return bool(not result.returncode)
+
+
+def clangformat(params):
+    """ Runs clang-format over the source folder excluding some optional folders.
+        It can also fix the errors found.
+
+    Args:
+        params: Uses 'params.output' to get the path to the build folder.
+                Uses 'params.source' to get the path to the source folder.
+                Uses 'params.exclude' to ignore folders from scan.
+                Uses 'params.fix' to optionally fix the errors found.
+
+    Returns:
+        bool: True on success, False otherwise.
+    """
+    file_extensions = ["*.cpp", "*.hpp"]
+    find_extensions = f'-iname {" -o -iname ".join(file_extensions)}'
+    if params.exclude:
+        abs_ignore = [os.path.join(params.source, path)
+                      for path in params.exclude]
+        find_ignoredir = f'-path {" -o -path ".join(abs_ignore)}'
+    find_cmd = f'find {params.source} -type f \( {find_extensions} \) -print -o \( {find_ignoredir} \) -prune'
+    clangformat_dry_cmd = 'clang-format --dry-run -Werror -style=file -i'
+    clangformat_cmd = 'clang-format -style=file -i'
+    logging.debug(f'Executing {find_cmd} | xargs {clangformat_dry_cmd}')
+    result = subprocess.run(f'{find_cmd} | xargs {clangformat_dry_cmd}',
+                            stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    if not result.stderr and not result.stdout:
+        logging.info('CLANG-FORMAT: successful')
+    else:
+        logging.info('CLANG-FORMAT: dry run fails.')
+        # TODO: we force the return code to 0 to allow the tool to continue
+        result.returncode = 0
+        if params.fix:
```

Comment:
It is expected that this condition will never be reached?

Reply:
The fix parameter was tested and it's working. 
Remember that `result.returncode = 0` only sets a variable, doesn't really returns yet.